### PR TITLE
Allow opening links in new tabs when HTML is trusted

### PR DESCRIPTION
**Pull request by yuvipanda**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/14919

## Code changes

When user opts to trust a HTML file in the HTML viewer, allow code there to open
pages in a new tab by adding `allow-popups` to the IFrame's sandbox attribute.

## User-facing changes

Once a user trusts a HTML file, the file can open new windows either via JS or
via `target="_blank"` links.

### CODE REVIEW

**Comment by fcollonval** [comment]
Thanks @yuvipanda 

I expanded following the discussion at #14919 to dynamically styled the links in the frame when it is untrusted. Let me know what do you think?

add_link_styling


cc @krassowski

**Comment by fcollonval** [comment]
Kicking CI following bot commit

File: packages/htmlviewer/src/widget.tsx [hunk 1]
```diff
@@ -33,6 +33,22 @@ const RENDER_TIMEOUT = 1000;
  */
 const CSS_CLASS = 'jp-HTMLViewer';
 
+const UNTRUSTED_LINK_STYLE = (options: { warning: string }) => `<style>
+a {
+  cursor: not-allowed !important;
+}
+a:hover::after {
```
Comment (krassowski) [inline packages/htmlviewer/src/widget.tsx:40]: Normal links work and do not need this. Only popup links are restricted.

```suggestion
a[target="_blank"] {
  cursor: not-allowed !important;
}
a[target="_blank"]:hover::after {
```

**Review by krassowski — COMMENTED** [review]
Thanks @yuvipanda and thank you @fcollonval - this solution is great!

File: packages/htmlviewer/src/widget.tsx (hunk 1 above)
Comment (fcollonval) [reply packages/htmlviewer/src/widget.tsx:40]: Thanks @krassowski
I was looking at the [documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes) and apparently 4 tags support `target`: `a`, `area`, `base` and `form`. I guess we should cover `a[target="_blank"]`, `area[target="_blank"]` and `form[target="_blank"]` (as the one for `base` is anyway overwritten) as the documentation for `allow-popups` does not specify any tag - only `target="blank_"`?

Comment (krassowski) [reply packages/htmlviewer/src/widget.tsx:40]: Right, thank you! I also see `formtarget`:

> This value can be overridden by a [formtarget](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#formtarget) attribute on a [`<button>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button), [`<input type="submit">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/submit), or [`<input type="image">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/image) element.

**Comment by fcollonval** [comment]
The latest commit 
- add the case of `formtarget`
- rephrase the warning to _Action disabled as the file is untrusted._ as button can become disabled; good English speaker feel free to suggest a better wording.

**Review by krassowski — APPROVED** [review]
Minor wording suggestions on tooltip.

**Comment by krassowski** [comment]
I wonder if "not trusted" would be more consistent. Also, we could set a non-serif font on the banner but it's all minor details.

**Comment by fcollonval** [comment]
CI failure is not related - thanks @krassowski for the review

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 4.0.x

**Pull request merged**